A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/MFD/A32NX_NDInfo.js

```diff
@@ -0,0 +1,531 @@
+          }
+          else if (delta < -180) {
+              endAngle += 360;
+          }
+          let smoothedAngle = Utils.SmoothSin(startAngle, endAngle, 0.25, this._dTime);
+          this.currentWindAngle = smoothedAngle % 360;
+          if (this.windDirection != null) {
+              if (windStrongEnough)
+                  this.windDirection.textContent = this.currentWindAngle.toFixed(0).padStart(3, "0");
+              else
+                  this.windDirection.textContent = "---";
+          }
+      }
+      if (refreshWindStrength) {
+          this.currentWindStrength = _windStrength;
+          if (this.windStrength != null) {
+              this.windStrength.textContent = this.currentWindStrength.toString().padStart(2, "0");
+          }
+      }
+      if (refreshWindArrow) {
+          this.currentPlaneAngle = _planeAngle;
+          if (this.windArrow != null) {
+              {
+                  var arrowAngle = this.currentWindAngle - this.currentPlaneAngle;
+                  arrowAngle += 180;
+                  var transformStr = this.windArrow.getAttribute("transform");
+                  var split = transformStr.split("rotate");
+                  if ((split != null) && (split.length > 0)) {
+                      this.windArrow.setAttribute("transform", split[0] + " rotate(" + arrowAngle + ")");
+                  }
+                  this.windArrow.style.display = "block";
+              }
+          }
+      }
+  }
+  setWaypoint(_name, _track, _distance, _eta, _force = false) {
+      if (this.waypoint) {
+          if (this._navMode == Jet_NDCompass_Navigation.NAV) {
+              if (_name && _name != "") {
+                  if (this.waypointName != null) {
+                      this.waypointName.textContent = _name;
+                  }
+                  if ((_track != this.currentWaypointTrack) || _force) {
+                      this.currentWaypointTrack = _track;
+                      if (this.waypointTrack) {
+                          this.waypointTrack.textContent = this.currentWaypointTrack.toString().padStart(3, "0") + String.fromCharCode(176);
+                      }
+                  }
+                  if ((_distance != this.currentWaypointDistance) || _force) {
+                      this.currentWaypointDistance = _distance;
+                      if (this.waypointDistance != null) {
+                          if (this.currentWaypointDistance < 10000)
+                              this.waypointDistance.textContent = this.currentWaypointDistance.toFixed(1);
+                          else
+                              this.waypointDistance.textContent = this.currentWaypointDistance.toFixed(0);
+                      }
+                  }
+                  if ((_eta != this.currentWaypointTimeETA) || _force) {
+                      this.currentWaypointTimeETA = _eta;
+                      const utcETA = (_eta + SimVar.GetGlobalVarValue("TIME ZONE OFFSET", "seconds")) % 86400;
```

Comment:
`const utcETA = (_eta + SimVar.GetGlobalVarValue("TIME ZONE OFFSET", "seconds") + 86400) % 86400;` should do as then the value never goes negative.

Reply:
Why would the value ever go negative?

Reply:
I ended up doing an entirely different approach that works better. I didn't think the new approach was possible since, at the time when I was first testing, it looked like waypoint ETE wasn't reporting correctly. I was just wrong and looking at the wrong simvar 😆